apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2Test.java

```diff
@@ -41,24 +65,1144 @@ public OutpatientClaim generateClaim() throws FHIRException {
+  }
+
+  @Test
+  public void shouldHaveLineItemProductOrServiceCoding() {
+    CodeableConcept pos = eob.getItemFirstRep().getProductOrService();
+
+    CodeableConcept compare =
+        new CodeableConcept()
+            .setCoding(
+                Arrays.asList(
+                    new Coding(
+                        "https://bluebutton.cms.gov/resources/codesystem/hcpcs", "M99", null)));
+
+    compare.setExtension(
+        Arrays.asList(
+            new Extension(
+                "http://hl7.org/fhir/sid/ndc",
+                new Coding("http://hl7.org/fhir/sid/ndc", "987654321", null))));
+
+    Assert.assertTrue(compare.equalsDeep(pos));
+  }
+
+  @Test
+  public void shouldHaveLineItemProductOrServiceExtension() {
+    Assert.assertNotNull(eob.getItemFirstRep().getProductOrService());
+    Assert.assertEquals(1, eob.getItemFirstRep().getProductOrService().getExtension().size());
+
+    Extension ex =
+        TransformerTestUtilsV2.findExtensionByUrl(
+            "http://hl7.org/fhir/sid/ndc",
+            eob.getItemFirstRep().getProductOrService().getExtension());
+
+    Extension compare =
+        new Extension(
+            "http://hl7.org/fhir/sid/ndc",
+            new Coding("http://hl7.org/fhir/sid/ndc", "987654321", null));
+
+    Assert.assertTrue(compare.equalsDeep(ex));
+  }
+
+  @Test
+  public void shouldHaveLineItemModifier() {
+    Assert.assertEquals(2, eob.getItemFirstRep().getModifier().size());
+
+    CodeableConcept modifier = eob.getItemFirstRep().getModifierFirstRep();
+
+    CodeableConcept compare =
+        new CodeableConcept()
+            .setCoding(
+                Arrays.asList(
+                    new Coding(
+                        "https://bluebutton.cms.gov/resources/codesystem/hcpcs", "XX", null)));
+
+    Assert.assertTrue(compare.equalsDeep(modifier));
+  }
+
+  @Test
+  public void shouldHaveLineItemServicedDate() {
+    DateType servicedDate = eob.getItemFirstRep().getServicedDateType();
+
+    DateType compare = new DateType("1942-01-03");
+
+    Assert.assertEquals(servicedDate.toString(), compare.toString());
+  }
+
+  @Test
+  public void shouldHaveLineItemLocation() {
+    Address address = eob.getItemFirstRep().getLocationAddress();
+
+    Address compare = new Address().setState("KY");
+
+    Assert.assertTrue(compare.equalsDeep(address));
+  }
+
+  @Test
+  public void shouldHaveLineItemQuantity() {
+    Quantity quantity = eob.getItemFirstRep().getQuantity();
+
+    Quantity compare = new Quantity(111);
+
+    Assert.assertTrue(compare.equalsDeep(quantity));
+  }
+
+  @Test
+  public void shouldHaveLineItemAdjudications() {
+    Assert.assertEquals(15, eob.getItemFirstRep().getAdjudication().size());
+  }
+
+  @Test
+  public void shouldHaveLineItemAdjudicationRevCntr1stAnsiCd() {
+    AdjudicationComponent adjudication =
+        TransformerTestUtilsV2.findAdjudicationByCategory(
+            "denialreason", eob.getItemFirstRep().getAdjudication());
```

Comment:
Might try to take a stab at one additional test tomorrow morning. There is a `rev_cntr_2nd_ansi_cd`, too. But the JSON looks like this for both:
```json
        {
          "category": {
            "coding": [
              {
                "system": "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBAdjudicationDiscriminator",
                "code": "denialreason",
                "display": "Denial Reason"
              }
            ]
          },
          "reason": {
            "coding": [
              {
                "system": "https://bluebutton.cms.gov/resources/variables/rev_cntr_1st_ansi_cd",
                "code": "CO120"
              }
            ]
          }
        }
```
The only existing helper looks for `denialreason` and can't come up with the second result. May write an additional helper, but wanted to get this out today.

Reply:
Added another util function to resolve this.